src/linear_algebra/bilinear_form.lean

```diff
@@ -1697,14 +1705,36 @@ theorem _root_.matrix.nondegenerate.to_bilin {M : matrix ι ι R₃} (h : M.nond
   (b : basis ι R₃ M₃) : (to_bilin b M).nondegenerate :=
 by convert h.to_bilin'.congr b.equiv_fun.symm using 1
 
+lemma _root_.matrix.nondegenerate_iff_to_bilin' {M : matrix ι ι R₃} :
+  M.nondegenerate ↔ (matrix.to_bilin' M).nondegenerate :=
+⟨matrix.nondegenerate.to_bilin', λ h v hv, h v $ λ w, (M.to_bilin'_apply' _ _).trans $ hv w⟩
```

Comment:
I guess we should probably have this for the unprimed `to_bilin` as well.

Reply:
It would also be good to have the very similar lemma:
```lean
B.to_matrix'.nondegenerate ↔ B.nondegenerate
```
which follows trivially from this one and the fact that `to_bilin` and `to_matrix` are inverses.

Finally, I think this probably makes sense as a `simp` lemma in the other direction?

```suggestion
@[simp] lemma _root_.matrix.nondegenerate_to_bilin'_iff {M : matrix ι ι R₃} :
  (matrix.to_bilin' M).nondegenerate ↔ M.nondegenerate :=
⟨λ h v hv, h v $ λ w, (M.to_bilin'_apply' _ _).trans $ hv w, matrix.nondegenerate.to_bilin'⟩
```